src/test/java/com/epam/rp/ui/pages/DashboardPage.java

```diff
@@ -48,13 +50,30 @@ public void selectFirstDashboard() {
         driver.findElement(firstDashboard).click();
     }
 
-    public void updateDashboardName(String newDashboardName) {
-        //wait.until(ExpectedConditions.visibilityOfElementLocated(editButton));
+    public void clickEditDashboardBtn(){
         wait.until(ExpectedConditions.elementToBeClickable(driver.findElement(editButton))).click();
+    }
+
+    public void updateDashboardName(String newDashboardName){
+        try{
+            clickEditDashboardBtn();
+        } catch (ElementClickInterceptedException ignored){
+        }
```

Comment:
It's better to avoid to build any logic based on exceptions. Maybe we may check some element's attribute or other element appearance to define necessity of the click action, but if it is really necessary it should be placed as close as possible to the call which may cause it.

E.g:

```java
        try{
             wait.until(ExpectedConditions.elementToBeClickable(driver.findElement(editButton))).click();
        } catch (ElementClickInterceptedException ignored){
          // The element maybe overlaed.
        }
```

Reply:
> It's better to avoid to build any logic based on exceptions. Maybe we may check some element's attribute or other element appearance to define necessity of the click action, but if it is really necessary it should be placed as close as possible to the call which may cause it.
> 
> E.g:
> 
> ```java
>         try{
>              wait.until(ExpectedConditions.elementToBeClickable(driver.findElement(editButton))).click();
>         } catch (ElementClickInterceptedException ignored){
>           // The element maybe overlaed.
>         }
> ```

Thank you for that advice, I agree with you, will fix that